typings.d.ts

```diff
@@ -0,0 +1,54 @@
+
+declare module "virgen-acl" {
+
+  export interface IResourceProp {
+    resource_id:string;
+  }
+
+  export interface IResourceGetter {
+    getResourceId():string;
+  }
+
+  export type IResource = IResourceGetter | IResourceProp | string
+
+  export interface IRoleProp {
+    role_id:string;
+  }
+
+  export interface IRoleGetter {
+    getRoleId():string;
```

Comment:
I don't know anything about type declarations, but this can also return an array.